setup/1. intellij.md

```diff
@@ -0,0 +1,29 @@
+# IntelliJ and MC Development
+
+## IntelliJ
+
+IntelliJ is an IDE created by JetBrains. 
+IDE stands for Integrated Development Environment, and is software designed to make programming as easy as possible. 
+Features that IDEs often have are ways to compile all the code with the press of a button and to help complete code with valid suggestions. 
+
+IntelliJ can be downloaded for all major operating systems from [IntelliJ Download](https://www.jetbrains.com/idea/download/) and has a free community version. The Ultimate version can be installed for free with a valid student account.
+
+While other Java IDEs can be used to create Fabric Mods, IntelliJ is the best IDE for it, as it has the Minecraft Development Plugin, and the integration with Gradle, a Java build tool that Fabric uses is better than other IDEs.
+
+### Installing Java
+
+If you have not installed Java onto your computer, or do not have Java 16 (needed for Minecraft 1.17+), you will need to install Java.
+At the time of writing the [Java 16 Hotspot](https://adoptopenjdk.net/?variant=openjdk16&jvmVariant=hotspot) is the version that you should install. 
+
+## Minecraft Development Plugin
+
+The Minecraft Developement Plugin, also known as MCDev is a community made plugin for IntelliJ that provides many useful error checking features for many different modding platforms. 
+
+To install the plugin, click on `File -> Settings` in IntelliJ. 
+A box will open and there should be a `Plugins` option on the left to be selected, then make sure that `Marketplace` and not `Installed` is selected on the top. 
+In the search bar, type `Minecraft` and MCDev should be the first plugin to show up. 
+Click install next to the name.
+
+### Issues
+
+MCDev is not perfect. It has to support a wide range of Minecraft versions and modding platforms, leading to some issues. While not super important for now, it will report an error if a custom `Entity` does not have a constructor that only takes in a `World` parameter. This can be ignored. In combination, do not use MCDev to create a project for you, as this often causes issues as the generated project is not configured correctly.
```

Comment:
We might want to state that you want to generate a project from [the example mod](https://github.com/FabricMC/fabric-example-mod) instead. That way they at least have a direction instead of just being told what not to do